core/editor/loader/v1/editor-v1-loader.php

```diff
@@ -0,0 +1,146 @@
+<?php
+namespace Elementor\Core\Editor\Loader\V1;
+
+use Elementor\Core\Editor\Loader\Common\Editor_Common_Scripts_Settings;
+use Elementor\Core\Editor\Loader\Editor_Base_Loader;
+use Elementor\Core\Utils\Assets_Config_Provider;
+use Elementor\Core\Utils\Collection;
+use Elementor\Plugin;
+use Elementor\Utils;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Editor_V1_Loader extends Editor_Base_Loader {
+	/**
+	 * @var Assets_Config_Provider
+	 */
+	protected $assets_config_provider;
+
+	/**
+	 * @param Collection $config
+	 */
+	public function __construct( Collection $config ) {
+		parent::__construct( $config );
+
+		$this->assets_config_provider = new Assets_Config_Provider();
+	}
+
+	/**
+	 * @return void
+	 */
+	public function init() {
+		parent::init();
+
+		$packages_to_register = [ 'ui', 'icons' ];
+
+		$assets_path = $this->config->get( 'assets-path' );
+
+		foreach ( $packages_to_register as $package ) {
+			$this->assets_config_provider->load(
+				$package,
+				"{$assets_path}js/packages/{$package}/{$package}.asset.php"
+			);
```

Comment:
Maybe the assets config provider can get a base path (or a resolver callback?) in the constructor and then each time you want to load you just:

```suggestion
			$this->assets_config_provider->load( $package );
```

Reply:
Lets talk about it

Reply:
Done